src/main/resources/mappings/workflow-steps.json

```diff
@@ -137,13 +137,7 @@
     "register_agent": {
         "inputs":[
             "name",
-            "type",
-            "tools",
-            "parameters",
-            "memory",
-            "created_time",
-            "last_updated_time",
-            "app_type"
```

Comment:
Catching up on these JSON files. Where do we persist the set of optional fields, besides `optionalKeys` in the `WorkflowStep` itself?

Reply:
Good question. There was talk of putting optional fields in this JSON file. There was also talk of getting rid of this file and creating an API to return all this stuff programmatically.  We have done neither.  

For now, though, this file is used for validation and these fields need to go.

We do need to figure out how else to document these.  See point 3 in https://github.com/opensearch-project/flow-framework/issues/315